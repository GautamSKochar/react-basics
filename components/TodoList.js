import React from 'react';

class TodoList extends React.Component {
  render() {
    return (
      <ul style={{ backgroundColor: 'lightgrey', maxWidth: '40%', marginBottom: '20px' }}>
        {this.props.items.map(item => (
          <li key={item.id}>{item.text}</li>
        ))}
      </ul>
    );
  }
}

export default TodoList;