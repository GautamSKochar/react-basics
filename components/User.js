import React from 'react';
import TodoList from './TodoList';
import UserList from './UserList';
import UserDetail from '../containers/UserDetails';

class User extends React.Component {
    constructor(props) {
        super(props);
        this.state = { items: [], text: '' };
        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
        this.onNavigateHome = this.onNavigateHome.bind(this);
    }
    onNavigateHome() {
        console.log('Button clicked, go to root');
        this.props.history.push('/');
    };
    render() {
        const { match, location, history } = this.props
        return (
            <div>
                <h3>The User Page</h3>
                <hr />
                <p>Hello {this.props.name}</p>
                <p>User Id: {this.props.match.params.id}</p>
                <button onClick={this.onNavigateHome} className="btn btn-primary">Go Home!</button>
                <div>
                    <h3>TODO</h3>
                    {/* <TodoList /> */}
                    <TodoList items={this.state.items} />
                    <form>
                        <input className="form-control col-sm-3"
                            onChange={this.handleChange}
                            value={this.state.text}
                        /><br />
                        <button type="button" onClick={this.handleSubmit} className="btn btn-primary">
                            Add #{this.state.items.length + 1}
                        </button>
                    </form>
                </div>
                <hr />
                <div>
                    <UserList />
                    <hr />
                    <UserDetail />
                </div>
            </div>
        );
    }

    handleChange(e) {
        this.setState({
            text: e.target.value
        });
    }

    handleSubmit(e) {
        if (!this.state.text.length) return;
        const newItem = {
            text: this.state.text,
            id: Date.now()
        };
        this.setState(prevState => ({
            items: prevState.items.concat(newItem),
            text: ''
        }));
    }
}
export default User;