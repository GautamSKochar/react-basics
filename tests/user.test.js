import React from 'react';
import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import User from '../components/User';
import TodoList from '../components/TodoList';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

const raf = global.requestAnimationFrame = (cb) => {
    setTimeout(cb, 0)
}
configure({ adapter: new Adapter() });

test('User Component Should render as expected', () => {
    const component = shallow(<User name="aaaabbbccc" />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
});

describe('User Todo List', () => {
    let list;
    beforeEach(() => {
        list = shallow(<TodoList items={['Name 1', 'Name 2', 'Name 3']} />);
    });

    it('List renders ul', () => {
        console.log("List renders ul element");
        expect(list.find('ul').length).toEqual(1);
    });

    it('List renders li elements', () => {
        expect(list.find('li').length).toEqual(3);
        console.log("List renders li elements");
    });

});