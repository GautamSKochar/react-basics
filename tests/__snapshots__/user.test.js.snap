// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Component Should render as expected 1`] = `
<div>
  <h3>
    The User Page
  </h3>
  <hr />
  <p>
    Hello 
    aaaabbbccc
  </p>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Go Home!
  </button>
  <div>
    <h3>
      TODO
    </h3>
    <TodoList
      items={Array []}
    />
    <form>
      <input
        className="form-control col-sm-3"
        onChange={[Function]}
        value=""
      />
      <br />
      <button
        className="btn btn-primary"
        onClick={[Function]}
        type="button"
      >
        Add #
        1
      </button>
    </form>
  </div>
</div>
`;
